include/seqan/blast/blast_report_out.h

```diff
@@ -0,0 +1,893 @@
+
+// ============================================================================
+// Functions
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// Function guessFormat()
+// ----------------------------------------------------------------------------
+
+template <typename TSpec>
+inline bool guessFormat(FormattedFile<BlastReport, Output, TSpec> &)
+{
+    return true;
+}
+
+// ----------------------------------------------------------------------------
+// Function _numberOfDigits
+// ----------------------------------------------------------------------------
+
+template <typename T>
+constexpr typename std::enable_if<std::is_integral<T>::value, T>::type
+_numberOfDigits(T const number)
+{
+    return (number == 0) ? 1 : static_cast<T>(std::floor(std::log10(number))+1);
+}
+
+// ----------------------------------------------------------------------------
+// some string constants
+// ----------------------------------------------------------------------------
+
+constexpr const char *
+_blastReference()
+{
+    return "Reference: Altschul, Stephen F., Thomas L. Madden, Alejandro "
+    "A. Schaffer,\nJinghui Zhang, Zheng Zhang, Webb Miller, and David J. "
+    "Lipman (1997),\n\"Gapped BLAST and PSI-BLAST: a new generation of protein "
+    "database search\nprograms\",  Nucleic Acids Res. 25:3389-3402.\n";
+}
+
+constexpr const char *
+_seqanReference()
+{
+    return "Reference for SeqAn: Doering, A., D. Weese, T. Rausch, K. Reinert (2008): "
+    "SeqAn --\nAn efficient, generic C++ library for sequence analysis. BMC "
+    "Bioinformatics,\n9(1), 11. BioMed Central Ltd."
+    " doi:10.1186/1471-2105-9-11\n";
+}
+
+template <typename T>
+constexpr const char *
+_matrixName(T const & /**/)
+{
+    return "'_matrixName not implemented for this matrix'";
+}
+
+constexpr const char *
+_matrixName(Blosum45 const & /**/)
+{
+    return "BLOSUM45";
+}
+
+constexpr const char *
+_matrixName(Blosum62 const & /**/)
+{
+    return "BLOSUM62";
+}
+
+constexpr const char *
+_matrixName(Blosum80 const & /**/)
+{
+    return "BLOSUM80";
+}
+
+constexpr const char *
+_matrixName(Pam250 const & /**/)
+{
+    return "PAM250";
+}
+
+// ----------------------------------------------------------------------------
+// Function _writeStatsBlock
+// ----------------------------------------------------------------------------
+
+template <typename TStream,
+          typename TScore,
+          typename TConString,
+          typename TMatch,
+          BlastProgram p,
+          BlastTabularSpec h>
+inline void
+_writeStatsBlock(TStream & stream,
+                 BlastIOContext<TScore, TConString, p, h> const &,
+                 TMatch const & m,
+                 BlastReport const &)
+{
+    char buffer[512] = "";
+
+    sprintf(buffer," Score =  %.1f bits (%d), Expect =  %.1g\n"
```

Comment:
Because it would take 3 pages of code to do the same by writing FormattedNumber :o